### PR TITLE
Draft attempt to fix yapf.yml

### PR DESCRIPTION
**Pull request by erogluorhan**
WIP:

Auto yapf commits (complete_yapf) was being pushed by "web-flow", which was likely the reason of automatic workflows not being triggered. This PR aims to fix it by signing such auto pushes with @pilotchute's github user name.

### CODE REVIEW

**Comment by andersy005** [comment]
@erogluorhan, because of how GitHub shares action secrets between a base repository and a fork, you may want to test this on a fork to confirm that things work as you'd expect for forks. I presume you want this to work for people with access to this repo and as well as for external contributors who don't have push permissions on this repository.

**Comment by erogluorhan** [comment]
> @erogluorhan, because of how GitHub shares action secrets between a base repository and a fork, you may want to test this on a fork to confirm that things work as you'd expect for forks. I presume you want this to work for people with access to this repo and as well as for external contributors who don't have push permissions on this repository.

That's absolutely correct! I was just going to ask you to try this change in your forked repo via CI update PR (since I think your forked repo is open to others' commits). Can you try this?

However, regarding general forked repos where @pilotchute may not have commit rights, I originally looked for a way to fetch the owner of the branch automatically but couldn't do that so far, the I decided to put @pilotchute at least as he was the implementer of yapf.

**Comment by andersy005** [comment]
> Auto yapf commits (complete_yapf) was being pushed by "web-flow", which was likely the reason of automatic workflows not being triggered.  

I think the workflow has been working fine. The reason the automatic workflows were not being triggered is because of this if condition

https://github.com/NCAR/geocat-comp/blob/eb5352209ff2ced8c1885ede7d60008dea5fc0c7/.github/workflows/yapf.yml#L8


which looks for the `trigger_yapf` string in the git commit summary line. The git commit message that was being used is 

https://github.com/NCAR/geocat-comp/blob/eb5352209ff2ced8c1885ede7d60008dea5fc0c7/.github/workflows/yapf.yml#L30

As you can see, this git message `complete_yapf` doesn't pass the if condition which requires the presence of `trigger_yapf`

**Comment by erogluorhan** [comment]
> > Auto yapf commits (complete_yapf) was being pushed by "web-flow", which was likely the reason of automatic workflows not being triggered.
> 
> I think the workflow has been working fine. The reason the automatic workflows were not being triggered is because of this if condition
> 
> https://github.com/NCAR/geocat-comp/blob/eb5352209ff2ced8c1885ede7d60008dea5fc0c7/.github/workflows/yapf.yml#L8
> 
> which looks for the `trigger_yapf` string in the git commit summary line. The git commit message that was being used is
> 
> https://github.com/NCAR/geocat-comp/blob/eb5352209ff2ced8c1885ede7d60008dea5fc0c7/.github/workflows/yapf.yml#L30
> 
> As you can see, this git message `complete_yapf` doesn't pass the if condition which requires the presence of `trigger_yapf`

This is correct, but my point is that after "complete_yapf" commit, even your auto workflow is not being triggered, which is based on `ci.yml` and does not have any if checks. I am aiming at having your `ci.yml` 100% triggering after any push

File: .github/workflows/yapf.yml [hunk 1]
```diff
@@ -27,10 +27,16 @@ jobs:
       - if: contains(github.event.head_commit.message, 'trigger_yapf')
         name: push_yapf
         run: |
+          git config --global user.name 'A. Kootz'
+          git config --global user.email 'pilotchute@users.noreply.github.com'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
```
Comment (pilotchute) [inline .github/workflows/yapf.yml:32]: @erogluorhan I am deeply uncomfortable with commits that use my name and token, from an automated system. I strongly do not want my name attached to commits that I do not make.

Comment (erogluorhan) [reply .github/workflows/yapf.yml:32]: Understood.

This is a draft PR to see if auto commit of yapf can be signed with a username, more precisely with the name of the branch's owner. I could not achieve that and just put yours as the developer of yapf.yml. And again, it is not intended to be finalized in this shape, I aimed at opening a work together here to sign them properly.

Comment (pilotchute) [reply .github/workflows/yapf.yml:32]: Ok, I didn't get that from context, and I was quite surprised to find a commit with my name on it, that I had not made.

**Pull request closed without merging**